core/src/main/scala/org/apache/spark/sql/delta/ConflictChecker.scala

```diff
@@ -96,8 +96,9 @@ private[delta] class WinningCommitSummary(val actions: Seq[Action], val commitVe
   val changedDataAddedFiles: Seq[AddFile] = if (isBlindAppendOption.getOrElse(false)) {
     Seq()
   } else {
-    addedFiles
+    addedFiles.filter(_.dataChange)
   }
+  val changedDataRemovedFiles: Seq[RemoveFile] = removedFiles.filter(_.dataChange)
```

Comment:
why not play it safe and perform the same `isBlindAppendOption.getOrElse(false)` as above?

Reply:
I think we should detect any removeFile when isBlinedAppendOption=true as it's not blind append apparently. What do you think ?

Reply:
That makes sense